src/zcbor_decode.c

```diff
@@ -274,14 +274,19 @@ static bool str_start_decode(zcbor_state_t *state,
 		ZCBOR_FAIL();
 	}
 
+	result->value = state->payload;
+	return true;
+}
+
+
+static bool str_overflow_check(zcbor_state_t *state, struct zcbor_string *result)
```

Comment:
```suggestion
static bool str_overflow_check(const zcbor_state_t *state, struct zcbor_string *result)
```
?

Reply:
The ERR_RESTORE() needs a mutable state.